paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/customersheet/CustomerSheetPlaygroundViewModel.kt

```diff
@@ -0,0 +1,242 @@
+import com.stripe.android.paymentsheet.example.Settings
+import com.stripe.android.paymentsheet.example.playground.customersheet.model.CreateSetupIntentRequest
+import com.stripe.android.paymentsheet.example.playground.customersheet.model.CreateSetupIntentResponse
+import com.stripe.android.paymentsheet.example.playground.customersheet.model.CustomerEphemeralKeyRequest
+import com.stripe.android.paymentsheet.example.playground.customersheet.model.CustomerEphemeralKeyResponse
+import com.stripe.android.paymentsheet.example.playground.customersheet.settings.CountrySettingsDefinition
+import com.stripe.android.paymentsheet.example.playground.customersheet.settings.CustomerSheetPlaygroundSettings
+import com.stripe.android.paymentsheet.example.playground.customersheet.settings.PaymentMethodMode
+import com.stripe.android.paymentsheet.example.playground.customersheet.settings.PaymentMethodModeDefinition
+import com.stripe.android.paymentsheet.example.samples.networking.awaitModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
+
+internal class CustomerSheetPlaygroundViewModel(
+    application: Application,
+    launchUri: Uri?,
+) : AndroidViewModel(application) {
+    private val settings by lazy {
+        Settings(application)
+    }
+
+    val playgroundSettingsFlow = MutableStateFlow<CustomerSheetPlaygroundSettings?>(null)
+    val playgroundState = MutableStateFlow<CustomerSheetPlaygroundState?>(null)
+
+    val status = MutableSharedFlow<StatusMessage>()
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
+            playgroundSettingsFlow.value =
+                CustomerSheetPlaygroundUrlHelper.settingsFromUri(launchUri)
+                    ?: CustomerSheetPlaygroundSettings.createFromSharedPreferences(application)
+        }
+    }
+
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    fun reset() {
+        val playgroundSettings = playgroundSettingsFlow.value ?: return
+
+        val setupIntentClientSecretProvider = if (
+            playgroundSettings[PaymentMethodModeDefinition].value == PaymentMethodMode.SetupIntent
+        ) {
+            ::createSetupIntentClientSecret
+        } else {
+            null
+        }
+
+        playgroundState.value = CustomerSheetPlaygroundState(
+            snapshot = playgroundSettings.snapshot(),
+            adapter = CustomerAdapter.create(
+                context = getApplication(),
+                customerEphemeralKeyProvider = ::fetchEphemeralKey,
+                setupIntentClientSecretProvider = setupIntentClientSecretProvider
+
+            ),
+            optionState = CustomerSheetPlaygroundState.PaymentOptionState.Unloaded
+        )
+    }
+
+    @OptIn(ExperimentalCustomerSheetApi::class)
+    fun fetchOption(customerSheet: CustomerSheet) {
```

Comment:
I took some time to dig into this. I think we did it this way because of the `PaymentOptionFactory`, and the only way we can supply that to create them is via `CustomerSheet`.

In hindsight, we likely could have hacked around that technical limitation, and made a more compelling public API. Maybe we should reconsider before GA.

`retrievePaymentOptionSelection` should probably live on the adapter. But how do we allow merchants to transform it to a `PaymentOption`? As of now, we don't have a way to do that.